src/Game/AccountManagement.java

```diff
@@ -0,0 +1,311 @@
+/**
+ * rscplus
+ *
+ * <p>This file is part of rscplus.
+ *
+ * <p>rscplus is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * <p>rscplus is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * <p>You should have received a copy of the GNU General Public License along with rscplus. If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ * <p>Authors: see <https://github.com/RSCPlus/rscplus>
+ */
+package Game;
+
+import java.math.BigInteger;
+import Client.CRC16;
+import Client.Logger;
+import Client.Util;
+
+public class AccountManagement {
+	
+	// § SECTION stream IO interaction §
+	
+	public static void register(String user, String pass) {
+		if (Client.login_delay > 0) {
+			Client.setLoginMessage("Connecting to server", "Please wait...");
+			try {
+				Client.shadowSleep(11200, 2000L);
+			} catch (Exception e) {}
+			Client.setLoginMessage("Please try again later", "Sorry! The server is currently full.");
+		} else {
+			try {
+				Client.username_login = user;
+				Client.password_login = pass;
+				String formatPass = Client.formatText(pass, 20);
+				Client.setLoginMessage("Connecting to server", "Please wait...");
+				//int port = Client.autologin_timeout <= 1 ? Client.serverjag_port : Replay.connection_port;
+				int port = Replay.connection_port;
+				StreamUtil.initializeStream(Client.server_address, port);
+				StreamUtil.setStreamMaxRetries(Client.maxRetries);
+				
+				StreamUtil.newPacket(2);
+                Object buffer = StreamUtil.getStreamBuffer();
+                StreamUtil.putShortTo(buffer, (short)(Client.version & 0xFFFF));
+
+                // Put Username
+                long formatUser = Util.username2hash(Client.username_login);
+                StreamUtil.putLongTo(buffer, formatUser);
+
+                // Put Password
+                enc_cred_put(buffer, formatPass, 0);
+
+                // In 235 putRandom of "random.dat" is 24 bytes, but for 127 is expected 4 bytes.
+                Object randBlock = StreamUtil.getNewBuffer(24);
+                StreamUtil.putRandom(randBlock);
+                byte[] randArr = StreamUtil.getBufferByteArray(randBlock);
+                CRC16 sum = new CRC16();
+                sum.update(randArr);
+                int rand = (int)sum.getValue();
+                StreamUtil.putIntTo(buffer, rand);
+
+                StreamUtil.flushPacket();
+                
+                int response = StreamUtil.readStream();				

```

Comment:
Above this line
`StreamUtil.readStream(); // Unknown what data this contained, client doesn't use it.`

Reply:
placed